grails-app/taglib/org/pih/warehouse/WordWrapTagLib.groovy

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) 2022 Partners In Health.  All rights reserved.
+ * The use and distribution terms for this software are covered by the
+ * Eclipse Public License 1.0 (http://opensource.org/licenses/eclipse-1.0.php)
+ * which can be found in the file epl-v10.html at the root of this distribution.
+ * By using this software in any fashion, you are agreeing to be bound by
+ * the terms of this license.
+ * You must not remove this notice, or any other, from this software.
+ */
+package org.pih.warehouse
+
+/**
+ * Methods for controlling word and line breaks when rendering tabular data.
+ */
+class WordWrapTagLib {
+
+    static String namespace = 'wordwrap'
+
+    /**
+     * Enclose `body` in an html element that wraps only on spaces.
+     *
+     * TIL you can't (currently) ask CSS to do this directly. Its default
+     * word-wrapping behavior allows breaks at slashes and hyphens,
+     * which can make a mess of tabular output containing dates and ID's.
```

Comment:
@jmiranda I couldn't find a way to tell CSS to wrap only on spaces, the comments above and below explain why in more detail. If you or @awalkowiak know a way to do this in CSS I will gladly delete this taglib. Thanks!

Reply:
Sorry, glossed right over the comment here.

Reply:
@mdpearson I can't think of and find anything that would be useful here. I think we have to leave this version.